libs/core/execution_base/include/hpx/execution_base/traits/coroutine_traits.hpp

```diff
@@ -0,0 +1,231 @@
+//  Copyright (c) 2022 Shreyas Atre
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <coroutine>
+#include <type_traits>
+
+namespace hpx { namespace execution { namespace experimental {
+
+    template <typename Promise, typename Awaiter>
+    decltype(auto) await_suspend(Awaiter&& await)
+    {
+        if constexpr (!std::is_same_v<Promise, void>)
+        {
+            return await.await_suspend(std::coroutine_handle<Promise>{});
+        }
+        return;
+    }
+
+    namespace detail {
+
+        template <typename T, typename... Ts>
+        bool inline constexpr one_of = (std::is_same_v<T, Ts> || ...);
+
+        template <class, template <class...> class>
+        constexpr bool is_instance_of_ = false;
+
+        template <class... _As, template <class...> class _T>
+        constexpr bool is_instance_of_<_T<_As...>, _T> = true;
+
+        template <class _Ty, template <class...> class _T>
+        constexpr bool is_instance_of = is_instance_of_<_Ty, _T>;
+
+        template <typename T>
+        inline constexpr bool is_await_suspend_result_t =
```

Comment:
We usually use the `_v` postfix for variables, not `_t`.

Reply:
Okay, noted